### PR TITLE
CI: Fix clipboard problems

### LINKED ISSUE #29676: CI: Clipboard Test Failures on Travis
```
[gw1] linux -- Python 3.6.6 /home/travis/miniconda3/envs/pandas-dev/bin/python
data = '👍...'
    @pytest.mark.single
    @pytest.mark.clipboard
    @pytest.mark.skipif(not _DEPS_INSTALLED, reason="clipboard primitives not installed")
    @pytest.mark.parametrize("data", ["\U0001f44d...", "Ωœ∑´...", "abcd..."])
    def test_raw_roundtrip(data):
        # PR #25040 wide unicode wasn't copied correctly on PY3 on windows
        clipboard_set(data)
>       assert data == clipboard_get()
E       AssertionError: assert '👍...' == ''
E         - 👍...
pandas/tests/io/test_clipboard.py:264: AssertionError
```
Examples:

https://travis-ci.org/pandas-dev/pandas/jobs/613141276?utm_medium=notification&utm_source=github_status

https://travis-ci.org/pandas-dev/pandas/builds/613153224?utm_source=github_status&utm_medium=notification

Potentially related change:
https://github.com/pandas-dev/pandas/pull/28531

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] closes #29676

Fixes the clipboard problems in the CI. With this PR we're sure they are being run, and they work as expected.

### CODE REVIEW

**Comment by datapythonista** [comment]
First run successful (problems uploading the coverage, but all tests passed in all builds). Restarting...

**Comment by datapythonista** [comment]
Or I don't think this should harm, if you prefer to merge and see in the other builds... Otherwise I'll restart the job couple times more to see if this looks like it's fixing the unreliable test.

**Comment by datapythonista** [comment]
Looks like this doesn't fix the problem. Re-reading again the docs, I think the new option would help errors if there are multiple `xvfb` processes, not if the wrapped command starts multiple processes.

I think it's still worth merging this, but still need to find out what's making that test fail.

**Comment by TomAugspurger** [comment]
@datapythonista maybe try https://github.com/pandas-dev/pandas/pull/29695/commits/6557e9b10f4e4138f674bff1788a56d109c66617 out? I didn't look into the failures for that run. Turns out they were unrelated.

I'm not sure how pytest-xdist splits up modules, but if it's done after importing the file, that first one would have already run.

**Comment by datapythonista** [comment]
> @datapythonista maybe try [6557e9b](https://github.com/pandas-dev/pandas/commit/6557e9b10f4e4138f674bff1788a56d109c66617) out?

Will have a look. `pytest-xdist` should allocate whole files to workers, I guess that will happen in the worker already, but not sure.

**Comment by TomAugspurger** [comment]
The whole file is definitely run on one worker (you can check with `-v`).
I'm just unsure whether or not the top-level of the module is executed
before being assigned to a worker.

**Comment by jbrockmendel** [comment]
could we put this into an allowed_fail or something so this doesn't hang up other PRs?  or would that make it too likely to slip through the cracks?

**Comment by TomAugspurger** [comment]
Pushed up the deferred check. I think if that's failing and @datapythonista is out of ideas then we can xfail these tests.

**Comment by jbrockmendel** [comment]
got green here.  does that mean this is fixed or just a lucky run?

**Comment by datapythonista** [comment]
Lucky run so far, let's try again...

**Review by TomAugspurger — COMMENTED** [review]
I don’t think we want this on the class. IUCC, the tests in the class have the clipboard mocked.

**Comment by datapythonista** [comment]
Yep, you're right. Not sure why I thought only the methods using `mock_clipboard` as a parameter. A bit strange that the only test that actually tests the clipboard is that last one, that seems to be added later to test the long unicode characters.

What I see is that `xsel` is only installed in an Azure build. In the failing build `qt` is indirectly installed by `conda`. I want to see were tests fail if they are not skipped. They are skipped if the clipboard doesn't work, and I guess it works when `xsel` or `qt` are found, but it's difficult to really tell (even more when conda can be installing them because of an unrelated dependency).

I'll continue tomorrow, it's 2am here, but once we know where they are running we may want to do one of:
- Install `xsel` there
- Remove `qt`

I'd also remove the skip, and add to the patterns `(not slow and not network and not clipboard)` in the builds where we don't want them to run. Not beautiful, but understanding what is going on with the magic of skips is very difficult. And I think it's a good fix enough for now, considering that we should be simplifying the tests soon in #29685.

File: pandas/tests/io/test_clipboard.py [hunk 1]
```diff
@@ -8,13 +8,7 @@
 from pandas import DataFrame, get_option, read_clipboard
 import pandas.util.testing as tm
 
-from pandas.io.clipboard import PyperclipException, clipboard_get, clipboard_set
```
Comment (jreback) [inline pandas/tests/io/test_clipboard.py]: I am not sure we can remove this; for a user this would make these tests fail if the deps are not installed.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/test_clipboard.py (hunk 1 above)
Comment (datapythonista) [reply pandas/tests/io/test_clipboard.py]: Good point.

Sorry the title was misleading, this started as a fix, but I'm more debugging now. For now I'm just trying to understand what's being executed, and what are the actual errors. I'm not sure we're only skipping the tests when we think we are. After I remove all the magic and we understand better the problem, we can discuss the fix. I'm not really sure what makes sense at this point.

Comment (jreback) [reply pandas/tests/io/test_clipboard.py]: sure happy to have you do that :->

Comment (jreback) [reply pandas/tests/io/test_clipboard.py]: let's just disable these directly disable these tests to clear the queue while you work on a patch. (separate PR is ok)

**Comment by TomAugspurger** [comment]
@datapythonista I've xfailed those tests on CI for now, so less urgency on this now. We'll want to get it sorted out before the release though.

**Comment by datapythonista** [comment]
@jreback @TomAugspurger this should fix the problems with the clipboard, and stop xfailing the tricky test.

What I'm doing here:
- Be explicit on when we run the tests (previously, we ignored the clipboard tests if the pandas `to_clipboard()` method was failing)
- Use `xvfb` when a X server is not available in the CI
- Install `xsel` in all the builds where we test the clipboard
- Uninstall `qt`, which was causing problems

**Comment by TomAugspurger** [comment]
Thanks for tracking this down. A bit unfortunate about qt, but uninstalling is probably the right move. I don’t expect this to cause issues elsewhere.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. I moved to the 1.0.0 milestone. @TomAugspurger if you are ok, merge away.

**Comment by TomAugspurger** [comment]
Thanks @datapythonista!

**Comment by mcepl** [comment]
The strange thing is that it seems this has not been fixed on i586. In openSUSE, test suite passes with x86_64, but fails with i586 on test_raw_rountrip:
```
[ 1652s] ______________________ test_raw_roundtrip[\U0001f44d...] _______________________
[ 1652s] [gw6] linux -- Python 3.8.3 /usr/bin/python3
[ 1652s] 
[ 1652s] data = '👍...'
[ 1652s] 
[ 1652s]     @pytest.mark.single
[ 1652s]     @pytest.mark.clipboard
[ 1652s]     @pytest.mark.parametrize("data", ["\U0001f44d...", "Ωœ∑´...", "abcd..."])
[ 1652s]     def test_raw_roundtrip(data):
[ 1652s]         # PR #25040 wide unicode wasn't copied correctly on PY3 on windows
[ 1652s]         clipboard_set(data)
[ 1652s] >       assert data == clipboard_get()
[ 1652s] E       AssertionError: assert '👍...' == ''
[ 1652s] E         + 👍...
[ 1652s] 
[ 1652s] ../../BUILDROOT/python-pandas-1.0.5-44.4.i386/usr/lib/python3.8/site-packages/pandas/tests/io/test_clipboard.py:256: AssertionError
```

[Complete logs of the failing build](https://github.com/pandas-dev/pandas/files/4851944/_log.zip)

**Comment by TomAugspurger** [comment]
Can you verify that the clipboard is set up properly on that system? If not then the clipboard tests need to be skipped.

**Comment by mcepl** [comment]
> Can you verify that the clipboard is set up properly on that system? If not then the clipboard tests need to be skipped.

1. What do you mean exactly by “properly set up clipboard”? This is chroot-based build system for openSUSE. We use ``xvfb-run`` to run tests in the fake X11 environment, but it is quite possible there is something wrong with it.

2. Couldn’t the test system somehow identify problems with clipboard settings?

**Comment by TomAugspurger** [comment]
We were erroneously skipping the clipboard tests so we decided to make these opt-out.

**Pull request merged**